SimPEG/EM/NSEM/ProblemNSEM.py

```diff
@@ -0,0 +1,599 @@
+
+            :param float freq: Frequency
+            :rtype: scipy.sparse.csr_matrix
+            :return: A
+        """
+
+        # Note: need to use the code above since in the 1D problem I want
+        # e to live on Faces(nodes) and h on edges(cells). Might need to rethink this
+        # Possible that _fieldType and _eqLocs can fix this
+        MeMui = self.MeMui
+        MfSigma = self.MfSigma
+        C = self.mesh.nodalGrad
+        # Make A
+        A = C.T*MeMui*C + 1j*omega(freq)*MfSigma
+        # Either return full or only the inner part of A
+        return A
+
+    def getADeriv(self, freq, u, v, adjoint=False):
+        """
+        The derivative of A wrt sigma
+        """
+
+        u_src = u['e_1dSolution']
+        dMfSigma_dm = self.MfSigmaDeriv(u_src)
+        if adjoint:
+            return 1j * omega(freq) * mkvc(dMfSigma_dm.T * v,)
+        # Note: output has to be nN/nF, not nC/nE.
+        # v should be nC
+        return 1j * omega(freq) * mkvc(dMfSigma_dm * v,)
+
+    def getRHS(self, freq):
+        """
+            Function to return the right hand side for the system.
+            :param float freq: Frequency
+            :rtype: numpy.ndarray (nF, 1), numpy.ndarray (nF, 1)
+            :return: RHS for 1 polarizations, primary fields
+        """
+
+        # Get sources for the frequncy(polarizations)
+        Src = self.survey.getSrcByFreq(freq)[0]
+        # Only select the yx polarization
+        S_e = mkvc(Src.S_e(self)[:, 1], 2)
+        return -1j * omega(freq) * S_e
+
+    def getRHSDeriv(self, freq, v, adjoint=False):
+        """
+        The derivative of the RHS wrt sigma
+        """
+
+        Src = self.survey.getSrcByFreq(freq)[0]
+
+        S_eDeriv = mkvc(Src.S_eDeriv_m(self, v, adjoint),)
+        return -1j * omega(freq) * S_eDeriv
+
+    def fields(self, m):
+        '''
+        Function to calculate all the fields for the model m.
+
+        :param np.ndarray (nC,) m: Conductivity model
+        '''
+        # Set the current model
+        self.curModel = m
+        # Make the fields object
+        F = self.fieldsPair(self.mesh, self.survey)
+        # Loop over the frequencies
+        for freq in self.survey.freqs:
+            if self.verbose:
+                startTime = time.time()
+                print('Starting work for {:.3e}'.format(freq))
+                sys.stdout.flush()
+            A = self.getA(freq)
+            rhs  = self.getRHS(freq)
+            Ainv = self.Solver(A, **self.solverOpts)
+            e_s = Ainv * rhs
+
+            # Store the fields
+            Src = self.survey.getSrcByFreq(freq)[0]
+            # NOTE: only store the e_solution(secondary), all other components calculated in the fields object
+            F[Src, 'e_1dSolution'] = e_s
+
+            if self.verbose:
+                print('Ran for {:f} seconds'.format(time.time()-startTime))
+                sys.stdout.flush()
+        return F
+
+# Note this is not fully functional.
```

Comment:
Yes, actually this should be done with `logging`. I want to have a look at implementing `logging`, since this is more of `Info` rather then `Warning`.